wsmaster/codenvy-hosted-api-license/src/main/java/com/codenvy/api/license/LicenseFilter.java

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * Checks system license conditions.
+ * @author Dmytro Nochevov
+ */
+@Singleton
+public class LicenseFilter implements Filter {
+    @Inject
+    protected RequestFilter          requestFilter;
+    @Inject
+    protected HttpJsonRequestFactory requestFactory;
+    @Inject
+    @Named("che.api")
+    protected String                 apiEndpoint;
+    @Inject
+    @Named("no.user.interaction")
+    protected boolean                noUserInteraction;
+
+    @Override
+    public void init(FilterConfig config) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        final HttpServletRequest httpRequest = (HttpServletRequest)request;
+        final HttpServletResponse httpResponse = (HttpServletResponse)response;
+        if (shouldSkip(httpRequest)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        try {
+            if (isFairSourceLicenseNotAccepted()) {
+                if (!noUserInteraction && isAdmin()) {
+                    sendUserToAcceptFairSourceLicensePage(httpResponse);
+                } else {
+                    createForbiddenAccessResponse(httpResponse);
+                }
+            } else {
+                chain.doFilter(request, response);
+            }
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
+
+    private boolean shouldSkip(HttpServletRequest request) {
+        return requestFilter.shouldSkip(request) || new UriStartFromRequestFilter(ImmutableList.of("/api/permissions",
+                                                                                                   "/api/user/settings",
+                                                                                                   "/api/license/legality",
+                                                                                                   "/api/license/fair-source-license")).shouldSkip(request);
+    }
+
+    private void createForbiddenAccessResponse(HttpServletResponse response) throws IOException {
+        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        response.setContentType(MediaType.APPLICATION_JSON);
+        try (PrintWriter writer = response.getWriter()) {
+            writer.write(DtoFactory.getInstance()
+                                   .toJson(new UnauthorizedException(CodenvyLicenseManager.FAIR_SOURCE_LICENSE_IS_NOT_ACCEPTED_MESSAGE).getServiceError()));
+        }
+    }
+
+    private boolean isAdmin() {
+        return EnvironmentContext.getCurrent().getSubject().hasPermission(SystemDomain.DOMAIN_ID, null, SystemDomain.MANAGE_CODENVY_ACTION);
+    }
+
+    private void sendUserToAcceptFairSourceLicensePage(HttpServletResponse response)
+        throws IOException {
+        UriBuilder redirectUrl = UriBuilder.fromPath("/site/auth/accept-fair-source-license");
```

Comment:
But `"/site/error/error-cookies-disabled"` is in configuration? no? https://github.com/codenvy/codenvy/blob/d5cc85019b44076f45d700ee327b7c044ed14cd8/assembly/onpremises-ide-packaging-war-ide/src/main/java/com/codenvy/onpremises/deploy/IdeModule.java#L38-L38